src/bentoml/_internal/runner/runner_handle/remote.py

```diff
@@ -270,3 +288,178 @@ async def is_ready(self, timeout: int) -> bool:
+
+    def __del__(self):
+        if self._loop is not None:
+            self._loop.call_soon_threadsafe(
+                lambda: asyncio.ensure_future(self._client.close())
+            )
+
+    @handle_triton_exception
+    async def get_model(
+        self, model_name: str
+    ) -> RunnerMethod[t.Any, P, tritongrpcclient.InferResult]:
+        if not await self._client.is_model_ready(model_name):
+            # model is not ready, try to load it
+            logger.debug("model '%s' is not ready, loading.", model_name)
+            await self._client.load_model(model_name)
+
+        method = RunnerMethod[t.Any, P, tritongrpcclient.InferResult](
+            runner=self.runner,
+            name=model_name,
+            # TODO: configuration for triton
+            config=RunnableMethodConfig(batchable=False, batch_dim=(0, 0)),
+            max_batch_size=0,
+            max_latency_ms=10000,
+        )
+
+        # setattr for given models
+        if model_name not in self.runner.__dict__:
+            object_setattr(self.runner, model_name, method)
+
+        return method
+
+    @handle_triton_exception
+    async def async_run_method(
+        self,
+        __bentoml_method: RunnerMethod[t.Any, P, tritongrpcclient.InferResult],
+        *args: P.args,
+        **kwargs: P.kwargs,
+    ) -> tritongrpcclient.InferResult:
+        from ..container import AutoContainer
+
+        assert (len(args) == 0) ^ (
+            len(kwargs) == 0
+        ), f"Inputs for model '{__bentoml_method.name}' can be given either as positional (args) or keyword arguments (kwargs), but not both. See https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#model-configuration"
+
+        model_name = __bentoml_method.name
+
+        # return metadata of a given model
+        model_metadata: pb.ModelMetadataResponse = (
+            await self._client.get_model_metadata(model_name)
+        )
+
+        pass_args = args if len(args) > 0 else kwargs
+        if len(model_metadata.inputs) != len(pass_args):
+            raise ValueError(
+                f"Number of provided arguments ({len(model_metadata.inputs)}) does not match the number of inputs ({len(pass_args)})"
+            )
+
+        input_params = Params["ext.NpNDArray"](*args, **kwargs).map(
+            AutoContainer.to_triton_payload
+        )
+
+        outputs = [
+            tritongrpcclient.InferRequestedOutput(output.name)
+            for output in model_metadata.outputs
+        ]
+        inputs: list[tritongrpcclient.InferInput] = []
+
+        if len(args) > 0:
+            for (infer_input, arg) in zip(model_metadata.inputs, input_params.args):
+                InferInput = tritongrpcclient.InferInput(
+                    infer_input.name,
+                    arg.shape,
+                    tritongrpcclient.np_to_triton_dtype(arg.dtype),
+                )
+                InferInput.set_data_from_numpy(arg)
```

Comment:
Why don't we just create `InferInput`s in the data containers? That feels more natural.